library/TileTypes.cpp

```diff
@@ -341,6 +341,11 @@ namespace DFHack
                 if (cur_direction == tileDirection(tt))
                     value |= 4;
 
+                // If the material is a plant, consider soil an acceptable alternative
+                if (cur_material == tiletype_material::PLANT and tileMaterial(tt) == tiletype_material::SOIL) {
```

Comment:
I kind of stuck with `and` back when I was a kid because I couldn't remember if the bitwise and was one or two &, and kinda never really let it go. Let me change that real quick.

Reply:
my goodness, I've been writing so much Lua recently that I didn't even notice that.